src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1836,47 +1836,99 @@ def setter(resource_group_name, vm_name, vm):
 
 
 # region VirtualMachines Images
-def list_vm_images(cmd, image_location=None, publisher_name=None, offer=None, sku=None,
-                   all=False):  # pylint: disable=redefined-builtin
+def list_vm_images(cmd, image_location=None, publisher_name=None, offer=None, sku=None, all=False, edge_zone=None):  # pylint: disable=redefined-builtin
     load_thru_services = all
 
     if load_thru_services:
-        if not publisher_name and not offer and not sku:
+        if not publisher_name and not offer and not sku and not edge_zone:
             logger.warning("You are retrieving all the images from server which could take more than a minute. "
-                           "To shorten the wait, provide '--publisher', '--offer' or '--sku'. Partial name search "
-                           "is supported.")
-        all_images = load_images_thru_services(cmd.cli_ctx, publisher_name, offer, sku, image_location)
+                           "To shorten the wait, provide '--publisher', '--offer' , '--sku' or '--edge-zone'."
+                           " Partial name search is supported.")
+        all_images = load_images_thru_services(cmd.cli_ctx, publisher_name, offer, sku, image_location, edge_zone)
     else:
-        all_images = load_images_from_aliases_doc(cmd.cli_ctx, publisher_name, offer, sku)
-        logger.warning(
-            'You are viewing an offline list of images, use --all to retrieve an up-to-date list')
+        if edge_zone is not None:
+            if not publisher_name and not offer and not sku:
+                logger.warning("You are retrieving all the images from server which could take more than a minute. "
+                               "To shorten the wait, provide '--publisher', '--offer' or '--sku'."
+                               "Partial name search is supported.")
+            all_images = load_images_thru_services(cmd.cli_ctx, publisher_name, offer, sku, image_location, edge_zone)
```

Comment:
I personally think that when edge_zone is not none, changing `load_thru_services` to `all` will be clearer and simpler than calling `load_images_thru_services()` again in the else branch

Reply:
Such as: `load_thru_services = all or edge_zone is not None`